react/week-2/lesson.md

```diff
@@ -193,270 +181,297 @@ We don't need to worry about changing the DOM ourselves! This is what makes Reac
+  );
+}
+```
 
-Container components usually have some state and handler methods. Because of this they must use the `class` syntax. Presentational components on the other hand don't require the more verbose syntax. Instead they usually use the functional syntax.
+Our component now has a `<button>`, which will call the `incrementCount` function when clicked:
 
-## React Hooks
+```js
+<button onClick={incrementCount}>Click me</button>
+```
 
-React is continously updated with features all the time, one that stands out the most are **React Hooks**, released in React v16.8.
+The `incrementCount` function then calculates the **new** state by adding 1 onto the current `count`. And then calls `setCount` to set the new state:
 
-### What are hooks?
+```js
+function incrementCount() {
+  const newState = count + 1;
+  setCount(newState);
+}
+```
 
-**Hooks** are functions that enable you manipulate the React state (and other lifecycle features, which you will be learning next week).
+`setCount` does two things. First, it updates the state that our component is "remembering". Whatever you pass as the argument to `setCount` will be remembered as the new state.
 
-### Why React hooks, over Class component state and methods?
+It also tells React that the old state that is **still shown in the DOM** is outdated and so the DOM needs to change. Because of this, React will re-render all of our components to figure out what to change in the DOM.
 
-- Code is less complex - being able to make your code more readable in the long term, make it more maintainable
-- Hooks are reusable - unlike methods that scoped within in a class components, you are able to create your own and reuse them elsewhere you in the code. Keeping your code DRY (Don't Repeat Yourself)
+This is the magic of React: we only need to tell React how we want to DOM to look and it figures out what to change.
 
+### Where Does State Live?
 
-### Counter Example - with hooks
+We have talked about how a component "remembers" state. In fact, each component remembers **separate** state from other components. This means we can have multiple different Counters, each with a different state ([interactive example](https://codesandbox.io/s/multiple-counters-xm1x4?file=/src/App.js)):
 
 ```js
-import React, { useState } from 'react';
+function App() {
+  return (
+    <div>
+      <Counter />
+      <Counter />
+      <Counter />
+    </div>
+  );
+}
+```
 
-function Counter() {
-  // Declare a new state variable, which we'll call "count"
+### Setting Multiple States
+
+So far we've only seen an example with one state variable. But you can create multiple state variables if you want! Let's see an example ([interactive example](https://codesandbox.io/s/multiple-state-variables-piq5w?file=/src/FruitCounter.js)):
+
+```js
+function FruitCounter() {
+  const [fruit, setFruit] = useState("bananas");
   const [count, setCount] = useState(0);
 
+  function handleOrangesClick() {
+    setFruit("oranges");
+  }
```

Comment:
Nit-picking here- lets add a new line after this function to separate it from the next one

Reply:
While in prod code I'd normally agree with you, this was intentional. I'm attempting to group the fruit handlers separate from the count handler.